cloudinit/config/cc_ssh.py

```diff
@@ -9,43 +9,26 @@
 """
 SSH
 ---
-**Summary:** configure ssh and ssh keys
+**Summary:** configure ssh and ssh keys (host and authorized)
 
-This module handles most configuration for ssh and ssh keys. Many images have
-default ssh keys, which can be removed using ``ssh_deletekeys``. Since removing
-default keys is usually the desired behavior this option is enabled by default.
+This module handles most configuration for ssh and both host and authorized ssh
+keys.
 
-Keys can be added using the ``ssh_keys`` configuration key. The argument to
-this config key should be a dictionary entries for the public and private keys
-of each desired key type. Entries in the ``ssh_keys`` config dict should
-have keys in the format ``<key type>_private`` and ``<key type>_public``, e.g.
-``rsa_private: <key>`` and ``rsa_public: <key>``. See below for supported key
-types. Not all key types have to be specified, ones left unspecified will not
-be used. If this config option is used, then no keys will be generated.
+Authorized Keys
+^^^^^^^^^^^^^^^
 
-.. note::
-    when specifying private keys in cloud-config, care should be taken to
-    ensure that the communication between the data source and the instance is
-    secure
-
-.. note::
-    to specify multiline private keys, use yaml multiline syntax
+Authorized keys are a list of public SSH key pairs that are allowed to
+connect to a system. Authorized keys for the default user/first user defined
+in ``users`` can be specified using ``ssh_authorized_keys``. Keys should be
+specified as a list of public keys.
 
-If no keys are specified using ``ssh_keys``, then keys will be generated using
-``ssh-keygen``. By default one public/private pair of each supported key type
-will be generated. The key types to generate can be specified using the
-``ssh_genkeytypes`` config flag, which accepts a list of key types to use. For
-each key type for which this module has been instructed to create a keypair, if
-a key of the same type is already present on the system (i.e. if
-``ssh_deletekeys`` was false), no key will be generated.
-
-Supported key types for the ``ssh_keys`` and the ``ssh_genkeytypes`` config
-flags are:
+Importing ssh public keys for the default user (defined in ``users``)) is
```

Comment:
I think this paragraph you've inherited from the existing docs is just plain wrong. It's the only mention of `allow_publish_ssh_keys` in the entire project.

Reply:
Agreed, I think I would normally say we could follow up, but I've removed it for now.